src/MongoDB.Driver.Core/Core/Misc/SemanticVersion.cs

```diff
@@ -129,6 +128,12 @@ public int CompareTo(SemanticVersion other)
                 return result;
             }
 
+            if (ServerVersion.TryParse(ToString(), out var serverVersion) &&
```

Comment:
can we use the new class inside this one? We cannot remove this class, but I think we can use the new class to handle all logic for this class?

Reply:
This isn't what we discussed. We were only going to delegate to ServerVersion if one or the other SemanticVersion was a server internal build: